digiroad2-api-oth/src/test/scala/fi/liikennevirasto/digiroad2/IntegrationApiSpec.scala

```diff
@@ -294,4 +305,143 @@ class IntegrationApiSpec extends FunSuite with ScalatraSuite with BeforeAndAfter
+      Property(1L, "main_sign_text", "", false, Seq(PropertyValue("test"))),
+      Property(1L, "structure", "", false, Seq(PropertyValue("1"))),
+      Property(1L, "condition", "", false, Seq(PropertyValue("1"))),
+      Property(1L, "size", "", false, Seq(PropertyValue("10"))),
+      Property(1L, "height", "", false, Seq(PropertyValue("10"))),
+      Property(1L, "coating_type", "", false, Seq(PropertyValue("1"))),
+      Property(1L, "sign_material", "", false, Seq(PropertyValue("2"))),
+      Property(1L, "location_specifier", "", false, Seq(PropertyValue("11"))),
+      Property(1L, "terrain_coordinates_x", "", false, Seq(PropertyValue("1"))),
+      Property(1L, "terrain_coordinates_y", "", false, Seq(PropertyValue("2"))),
+      Property(1L, "lane_type", "", false, Seq(PropertyValue("1"))),
+      Property(1L, "lane", "", false, Seq(PropertyValue("1", propertyDisplayValue = Some("1")))),
+      Property(1L, "main_sign_text", "", false, Seq(PropertyValue("test"))),
+      Property(1L, "life_cycle", "", false, Seq(PropertyValue("1"))),
+      Property(1L, "trafficSign_start_date", "", false, Seq(PropertyValue("12.12.2000"))),
+      Property(1L, "trafficSign_end_date", "", false, Seq(PropertyValue("24.12.2000"))),
+      Property(1L, "type_of_damage", "", false, Seq(PropertyValue("2"))),
+      Property(1L, "urgency_of_repair", "", false, Seq(PropertyValue("1"))),
+      Property(1L, "lifespan_left", "", false, Seq(PropertyValue("1", propertyDisplayValue = Some("1")))),
+      Property(1L, "additional_panel", "", false, Seq(AdditionalPanel(1, "testInfo", "testValue", 1, "testText", 1, 1, 1)))
+    )
+
+    val trafficSignForApi = PersistedTrafficSign(1L, 100L, 11.11, 22.22, 33.33, false, 0L, 235, properties, None, None,
+      None, None, 1, Some(1), LinkGeomSource.Unknown, false)
+
+    val trafficSignValuesFromApi = integrationApi.trafficSignsToApi(Seq(trafficSignForApi)).head
+    trafficSignValuesFromApi.get("typeOfDamage").get.getClass.getTypeName should be("java.lang.Integer")
+    trafficSignValuesFromApi.get("oldTrafficCode").get.getClass.getTypeName should be("java.lang.Integer")
+    trafficSignValuesFromApi.get("size").get.getClass.getTypeName should be("java.lang.Integer")
+    trafficSignValuesFromApi.get("height").get.getClass.getTypeName should be("java.lang.Integer")
+    trafficSignValuesFromApi.get("lane").get.getClass.getTypeName should be("java.lang.Integer")
+    trafficSignValuesFromApi.get("structure").get.getClass.getTypeName should be("java.lang.Integer")
+    trafficSignValuesFromApi.get("condition").get.getClass.getTypeName should be("java.lang.Integer")
+    trafficSignValuesFromApi.get("coatingType").get.getClass.getTypeName should be("java.lang.Integer")
+    trafficSignValuesFromApi.get("urgencyOfRepair").get.getClass.getTypeName should be("java.lang.Integer")
+    trafficSignValuesFromApi.get("lifespanLeft").get.getClass.getTypeName should be("java.lang.Integer")
```

Comment:
hyym eikös Int kelpaa

toInt vaikuttaa myös menevän java.lang.Integer.parseInt(toString) jossa se muunnetaan java Int

Reply:
https://stackoverflow.com/questions/8561898/how-to-do-an-instanceof-check-with-scalatest tuolta löytyy jotakin vähän scala maisempia tapoja testata

Reply:
ok, nyt on vähän eri tavalla